script.rb

```diff
@@ -0,0 +1,152 @@
+module Enumerable
+  # -----------my_each methods--------------
+
+  def my_each
+    return to_enum unless block_given?
+
+    index = 0
+    arr = to_a
+    while index < arr.length
+      yield(arr[index])
+      index += 1
+    end
+    self
+  end
+
+  # -----------my_each_with_index----------------------
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    arr = to_a
+    index = 0
+    while index < arr.length
+      yield(arr[index], index)
+      index += 1
+    end
+    self
+  end
+
+  # -----------my_select----------------------
+
+  def my_select
+    return to_enum unless block_given?
+
+    selected = []
+    arr = to_a
+    arr.my_each { |x| selected.push(x) if yield(x) }
+    selected
+  end
+
+  #--------------------my_all-----------------------
+  def my_all?(args = nil)
+    if args.nil? == false && args.is_a?(Class) == false
+      if args.is_a?(Regexp)
+        my_each { |i| return false unless !i[args].nil? || i[args] == 1 } # if Regexp
+      else
+        my_each { |i| return false unless i == args }
+      end
+    elsif args.is_a?(Class)
+      my_each { |i| return false unless i.is_a?(args) }
+
+    # if not Regexp
+
+    elsif args.nil? && block_given?
+      my_each { |i| return false unless yield i }
+    else
+      my_each { |i| return false if i.nil? || i == false }
+    end
+    true
+  end
+  #-------------------my_any?-----------------------
+
+  def my_any?(args = nil)
+    if args.nil? == false && args.is_a?(Class) == false
+      if args.is_a?(Regexp)
+        my_each { |i| return true if !i[args].nil? || i[args] == 1 } # if Regexp
+      else
+        my_each { |i| return true if i == args }
+      end
+    elsif args.is_a?(Class)
+      my_each { |i| return true if i.is_a?(args) }
+
+    # if not Regexp
+
+    elsif args.nil? && block_given?
+      my_each { |i| return true if yield i }
+    else
+      my_each { |i| return false unless i.nil? || i == false }
+    end
+    false
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any? 

- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil

```ruby     
[nil, false, true, []].my_any? == [nil, false, true, []].any? => true
```
- Yours returns the wrong answer
![image](https://user-images.githubusercontent.com/52209067/93888097-c8af7700-fca4-11ea-9a3b-261443c3183b.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689425-3f262300-6f67-11ea-88a9-2885ebab08d4.png)